src/main/java/duke/logic/commands/UpdateWeightCommand.java

```diff
@@ -71,18 +71,23 @@ public void stage0(User user, Storage storage) {
                 ui.showWeightUpdate(user, Integer.parseInt(weight), currentDate);
                 ui.showLine();
             } catch (NumberFormatException e) {
-                ui.showMessage(e.getMessage());
+                ui.showMessage("Please input a proper number for weight");
```

Comment:
Good job adding exception handling. Perhaps you could be more specific with the error message? Proper number may not be clear for new users. Maybe something like: "<user input> is not a valid weight. Please input your weight as a positive number". Also, do you check for negative weight inputs?